File: 3163-string-compression-iii/3163-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string ans = "";
        for (int i = 0; i < word.length(); ) {
            char c = word[i];
            int j = i, cnt = 0;
            while (j < word.length() && word[j] == c && cnt < 9) {
                cnt++;
                j++;
            }
            ans += to_string(cnt);
            ans.push_back(c);
            i = j;
        }
        return ans;
    }
};

File: 3163-string-compression-iii/3163-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        int n = word.length(), count = 0, i = 0, j = 0;
        StringBuilder ans = new StringBuilder();
        while (j < n) {
            count = 0;
            while (j < n && word.charAt(i) == word.charAt(j) && count < 9) {
                j++;
                count++;
            }
            ans.append(count).append(word.charAt(i));
            i = j;
        }
        return ans.toString();
    }
}